lib/tasks/duplicates_metadata.rake

```diff
@@ -5,30 +5,15 @@ namespace :decidim do
     desc "Merge all authorization encrypted metadata to decrypted metadata in user extended data"
     task metadata: :environment do
       logger = Logger.new($stdout)
-      logger.info(logger_output("Retrieving all extended socio demographic authorizations"))
-
-      authorizations = Decidim::Authorization.where(name: "extended_socio_demographic_authorization_handler")
-      updated = []
-      authorizations.each do |auth|
-        if auth.user.blank? || !auth.user.respond_to?(:extended_data)
-          logger.error(logger_output("Undefined user for authorization ID/#{auth.id}"))
-          next
-        end
-        next if auth.user.extended_data.include?("socio_postal_code")
-        next if auth.user.extended_data.include?("socio_city")
-        next if auth.user.extended_data.include?("socio_email")
-        next if auth.user.extended_data.include?("socio_phone_number")
-
-        extended_data = auth.metadata.deep_transform_keys { |key| "socio_#{key}" }
-
-        if auth.user.update(extended_data: auth.user.extended_data.merge(extended_data))
-          logger.info(logger_output("Updating user (ID/#{auth.user.id})"))
-          updated << auth.user.id
-        else
-          logger.error(logger_output("Errors happened while updating user (ID/#{auth.user.id})"))
-        end
-      end
-      logger.info(logger_output("Found #{authorizations.count} extended socio demographic authorizations"))
+      logger.info(logger_output("Retrieving all extended socio demographic and data authorizations"))
+
+      socio_authorizations = Decidim::Authorization.where(name: "extended_socio_demographic_authorization_handler")
```

Comment:
Isn't it possible to fetch all `extended` and `data` in a unique array and process them based on their name